packages/core/src/bases/chain-wallet.ts

```diff
@@ -40,5 +90,37 @@ export abstract class ChainWalletBase<
     await this.update();
   }
 
+  get stargateClient(): Promise<SigningStargateClient | undefined> {
+    const fn = async () => {
+      const rpcEndpoint = await this.rpcEndpoint;
```

Comment:
very cool! didn't know you can do async getters! 🙌🏻

Reply:
🙌🏻